oprf/oprf.go

```diff
@@ -0,0 +1,55 @@
+package oprf
+
+import (
+	"github.com/cloudflare/circl/oprf/group"
+)
+
+// BlindToken corresponds to a token that has been blinded.
+// Internally, it is a serialized Point/Element.
+type BlindToken []byte
+
+// IssuedToken corresponds to a token that has been issued.
+// Internally, it is a serialized Point/Element.
+type IssuedToken []byte
+
+// Token is the object issuance of the protocol
+type Token struct {
+	data  []byte
+	blind *group.Scalar
+}
+
+// Evaluation corresponds to the evaluation over a token
+type Evaluation struct {
+	element []byte
+}
+
+// PublicKey is a struct representing a public key
+type PublicKey *group.Point
+
+// PrivateKey is a struct representing a private key
+type PrivateKey *group.Scalar
+
+// Client is
+type Client interface {
+	// SetUp configures a new client by generating its context and keys
+	SetUp(group.CiphersuiteID) (string, PrivateKey)
+
+	// Blind generates a token and blinded data.
+	Blind(in []byte) (*Token, *BlindToken)
+
+	// Unblind unblinds the server response.
+	Unblind(pk *PublicKey, t *Token, b *BlindToken, e *Evaluation) *IssuedToken
+}
+
+// Server is
+// TODO: should we add: VerifyFinalize and EvaluateBatch?
+type Server interface {
+	// SetUp configures a new server by generating its context and keys
+	SetUp(group.CiphersuiteID) (string, PrivateKey)
+
+	// Evaluate evaluates the token
+	Evaluate(sk *PrivateKey, pk *PublicKey, b *BlindToken, e *Evaluation) *Evaluation
+
+	// Finalize outputs a byte array that corresponds to its input.
+	Finalize(t *Token, issuedT *IssuedToken, info []byte) []byte
```

Comment:
This seems to be missing `FullEvaluate` (though I don't think that's needed right now).

Reply:
Yes, it is. As this was summited prior to the discussion on the chat. I'll wait to put it there once it definitely lands on the draft.